src/Xrefcheck/Scan.hs

```diff
@@ -68,14 +67,11 @@ data ExclusionConfig' f = ExclusionConfig
 
 makeLensesWith postfixFields ''ExclusionConfig'
 
--- | File extension, dot included.
-type Extension = String
-
 -- | Way to parse a file.
 type ScanAction = CanonicalPath -> IO (FileInfo, [ScanError 'Parse])
 
 -- | All supported ways to parse a file.
-type FormatsSupport = Extension -> Maybe ScanAction
+type FileSupport = CanonicalPath -> IO $ Maybe ScanAction
 
```

Comment:
Instead of determining which scanner to use just from the file extension, I changed it to a more general check. Another option is to put the symlink scanner in a 'different level' that is above extension check: if it is a symlink, then symlink scanner, else get scanner from extension.

Reply:
How about the type `Bool -> Extension -> Maybe ScanAction`, where the first bool denotes `isLink`?

Reply:
I like it. It is less general and can be better understood 👍